### PR TITLE
chore: running some chores for notation-go

### LINKED ISSUE #287: Update references from Notary v2 to Notation
This repo is Notation library, we should update references from Notary v2 to Notation.
Here is a search `Notary v2` in this repo https://github.com/notaryproject/notation-go/search?q=%22Notary+v2%22&type=code

Fix the references in code only

### PR DESCRIPTION
**Pull request by Two-Hearts**
In this PR:
1. According to issue #287, updated `Notary v2` to `Notation`.
2. Updated error messages of trust policy to improve user experience. Related issue: https://github.com/notaryproject/notation/issues/621

Resolves #287

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 [![codecov](https://codecov.io/gh/notaryproject/notation-go/branch/main/graph/badge.svg)](https://codecov.io/gh/notaryproject/notation-go)
 [![Go Reference](https://pkg.go.dev/badge/github.com/notaryproject/notation-go.svg)](https://pkg.go.dev/github.com/notaryproject/notation-go@main)
 
-A collection of libraries for supporting Notation sign, verify, push, pull of oci artifacts. Based on Notary V2 standard.
+A collection of libraries for supporting Notation sign, verify, push, pull of oci artifacts. Based on Notation standard.
```
Comment (yizha1) [inline README.md:7]: Notary Project

File: verifier/trustpolicy/trustpolicy.go [hunk 2]
```diff
@@ -368,7 +371,7 @@ func (t *TrustPolicy) clone() *TrustPolicy {
 }
 
 // validateTrustStore validates if the policy statement is following the
-// Notary V2 spec rules for truststores
+// Notation spec rules for truststores
```
Comment (yizha1) [inline verifier/trustpolicy/trustpolicy.go:374]: Notary Project

File: verifier/trustpolicy/trustpolicy.go [hunk 3]
```diff
@@ -387,7 +390,7 @@ func validateTrustStore(statement TrustPolicy) error {
 }
 
 // validateTrustedIdentities validates if the policy statement is following the
-// Notary V2 spec rules for trusted identities
+// Notation spec rules for trusted identities
```
Comment (yizha1) [inline verifier/trustpolicy/trustpolicy.go:393]: Notary Project

File: verifier/trustpolicy/trustpolicy_test.go [hunk 4]
```diff
@@ -518,7 +519,7 @@ func TestApplicableTrustPolicy(t *testing.T) {
 
 	// non-existing Registry Scope
 	policy, err = (&policyDoc).GetApplicableTrustPolicy("non.existing.scope/repo@sha256:hash")
-	if policy != nil || err == nil || err.Error() != "artifact \"non.existing.scope/repo@sha256:hash\" has no applicable trust policy" {
+	if policy != nil || err == nil || err.Error() != "artifact \"non.existing.scope/repo@sha256:hash\" has no applicable trust policy. Trust policy applicability for a given artifact is determined by registryScopes. To create Notation trust policy: https://notaryproject.dev/docs/quickstart/#create-a-trust-policy" {
```
Comment (yizha1) [inline verifier/trustpolicy/trustpolicy_test.go:522]: To create a trust policy, see: https://notaryproject.dev/docs/quickstart/#create-a-trust-policy.

File: verifier/trustpolicy/trustpolicy_test.go [hunk 5]
```diff
@@ -545,7 +546,7 @@ func TestLoadDocument(t *testing.T) {
 	tempRoot := t.TempDir()
 	dir.UserConfigDir = tempRoot
 	_, err := LoadDocument()
-	if err == nil || err.Error() != fmt.Sprintf("trust policy is not present, please create trust policy at %s", filepath.Join(dir.UserConfigDir, dir.PathTrustPolicy)) {
+	if err == nil || err.Error() != fmt.Sprintf("trust policy is not present. To create Notation trust policy: %s", trustPolicyLink) {
```
Comment (yizha1) [inline verifier/trustpolicy/trustpolicy_test.go:549]: To create a trust policy, see: %s

File: verifier/trustpolicy/trustpolicy.go [hunk 6]
```diff
@@ -279,7 +282,7 @@ func LoadDocument() (*Document, error) {
 	policyDocument := &Document{}
 	err = json.NewDecoder(jsonFile).Decode(policyDocument)
 	if err != nil {
-		return nil, errors.New("malformed trustpolicy.json file")
+		return nil, fmt.Errorf("malformed trust policy. To create Notation trust policy: %s", trustPolicyLink)
```
Comment (yizha1) [inline verifier/trustpolicy/trustpolicy.go:285]: To create a trust policy, see: %s

File: verifier/trustpolicy/trustpolicy.go [hunk 7]
```diff
@@ -268,7 +271,7 @@ func LoadDocument() (*Document, error) {
 	if err != nil {
 		switch {
 		case errors.Is(err, os.ErrNotExist):
-			return nil, fmt.Errorf("trust policy is not present, please create trust policy at %s", filepath.Join(dir.UserConfigDir, dir.PathTrustPolicy))
+			return nil, fmt.Errorf("trust policy is not present. To create Notation trust policy: %s", trustPolicyLink)
```
Comment (yizha1) [inline verifier/trustpolicy/trustpolicy.go:274]: To create a trust policy, see: %s

File: verifier/trustpolicy/trustpolicy.go [hunk 8]
```diff
@@ -258,7 +261,7 @@ func (trustPolicyDoc *Document) GetApplicableTrustPolicy(artifactReference strin
 	} else if wildcardPolicy != nil {
 		return wildcardPolicy, nil
 	} else {
-		return nil, fmt.Errorf("artifact %q has no applicable trust policy", artifactReference)
+		return nil, fmt.Errorf("artifact %q has no applicable trust policy. Trust policy applicability for a given artifact is determined by registryScopes. To create Notation trust policy: %s", artifactReference, trustPolicyLink)
```
Comment (yizha1) [inline verifier/trustpolicy/trustpolicy.go:264]: To create a trust policy, see: %s

**Review by yizha1 — COMMENTED** [review]
Thanks @Two-Hearts. I left a few comments

File: README.md (hunk 1 above)
Comment (yizha1) [reply README.md:7]: If notation is also an implementation of Notary project specs, so if it is about following the specs, then suggest using Notary project, if it is pure notation, then `Notation` is more approporate.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**